### PR TITLE
feat: add keyboard shortcuts

### PR DESCRIPTION
**Pull request by AyushAgrawal-A2**
part of #1536 

* [x]  Default JSON for shortcut default theme
* [x]  F2: edit cell
* [x]  Support both Ctrl + Shift + PageUp / Ctrl + Shift + PageDown **AND** option + up-arrow, option + down-arrow **AND** option + right arrow/option + left arrow: go to next/prior sheet (this covers as close as possible to all bases for both mainstream spreadsheets)
* [x]  Ctrl + d: fill down from the cell above
* [x]  Ctrl + r: fill right from the cell to the left
* [x]  Shift + space: select row
* [x]  Ctrl + space: select column

### CODE REVIEW

**Review by davidfig — APPROVED** [review]

**Comment by davidfig** [comment]
Code looks good

File: quadratic-client/src/app/theme/shortcuts.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
+/**
+ * Default shortcuts for the application.
+ * Mac key names are used for the combinations.
+ */
+export const defaultShortcuts = {
+  // QuadraticGrid.tsx
+  grid_pan_mode: ['Space'],
+
+  // keyboardViewport.ts
+  show_command_palette: ['Cmd + P', 'Cmd + P', 'Cmd + /'],
```
Comment (davidkircos) [inline quadratic-client/src/app/theme/shortcuts.ts:10]: duplicate

File: quadratic-client/src/app/theme/shortcuts.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
+/**
+ * Default shortcuts for the application.
+ * Mac key names are used for the combinations.
+ */
+export const defaultShortcuts = {
```
Comment (davidkircos) [inline quadratic-client/src/app/theme/shortcuts.ts:5]: I think that our shortcut Schema should be typed with Zod, and should have platform specific options:

We may also want to add modifiers of when they are active: Check out VSCode keybindings schema 
https://code.visualstudio.com/docs/getstarted/keybindings#_keyboard-rules

Comment (davidkircos) [reply quadratic-client/src/app/theme/shortcuts.ts:5]: They also have a way of making shortcuts keyboard layout independent 
https://code.visualstudio.com/docs/getstarted/keybindings#_keyboard-layoutindependent-bindings

Comment (davidfig) [reply quadratic-client/src/app/theme/shortcuts.ts:5]: I like this approach.

**Comment by AyushAgrawal-A2** [comment]
moved to quadratic repo from fork #1612

File: quadratic-client/src/app/theme/shortcuts.ts (hunk 2 above)
Comment (AyushAgrawal-A2) [reply quadratic-client/src/app/theme/shortcuts.ts]: done

**Pull request closed without merging**